app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala

```diff
@@ -0,0 +1,1110 @@
+    implicit request =>
+      withFillingOutReturnAndYTDLiabilityAnswers(request) {
+        case (_, fillingOutReturn, answers) =>
+          (answers, fillingOutReturn.draftReturn) match {
+            case (calculatedAnswers: CalculatedYearToDateLiabilityAnswers, draftReturn: SingleDisposalDraftReturn) =>
+              withTaxDueAndCalculatedTaxDue(calculatedAnswers) { (taxDue, calculatedTaxDue) =>
+                if (calculatedTaxDue.amountOfTaxDue === taxDue) {
+                  Redirect(routes.YearToDateLiabilityController.checkYourAnswers())
+                } else {
+                  lazy val backLink = calculatedAnswers.fold(
+                    _ => routes.YearToDateLiabilityController.taxDue(),
+                    _ => routes.YearToDateLiabilityController.checkYourAnswers()
+                  )
+                  mandatoryEvidenceForm
+                    .bindFromRequest()
+                    .fold(
+                      formWithErrors => BadRequest(mandatoryEvidencePage(formWithErrors, backLink)), { s =>
+                        val updatedDraftReturn = draftReturn.copy(
+                          yearToDateLiabilityAnswers = Some(
+                            calculatedAnswers.fold(
+                              _.copy(mandatoryEvidence = Some(s)),
+                              _.copy(mandatoryEvidence = Some(s))
+                            )
+                          )
+                        )
+
+                        val result =
+                          for {
+                            _ <- returnsService.storeDraftReturn(
+                                  updatedDraftReturn,
+                                  fillingOutReturn.subscribedDetails.cgtReference,
+                                  fillingOutReturn.agentReferenceNumber
+                                )
+                            _ <- EitherT(
+                                  updateSession(sessionStore, request)(
+                                    _.copy(journeyStatus = Some(fillingOutReturn.copy(draftReturn = updatedDraftReturn))
+                                    )
+                                  )
+                                )
+                          } yield ()
+
+                        result.fold(
+                          { e =>
+                            logger.warn("Could not update return", e)
+                            errorHandler.errorResult()
+                          },
+                          _ => Redirect(routes.YearToDateLiabilityController.checkYourAnswers())
+                        )
+                      }
+                    )
+
+                }
+              }
+
+            case _ =>
+              Redirect(routes.YearToDateLiabilityController.checkYourAnswers())
+
+          }
+      }
+  }
+
+  def taxableGainOrLoss(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
```

Comment:
this is the only new endpoint for non-calculated journeys so far - it moved from the exemptions and losses section